projects/packages/account-protection/src/class-password-detection.php

```diff
@@ -225,38 +241,17 @@ public function render_page() {
 	 * @return void
 	 */
 	public function render_content( \WP_User $user, string $token ): void {
+		defined( 'ABSPATH' ) || exit;
+
 		$error_transient_key   = Config::PASSWORD_DETECTION_TRANSIENT_PREFIX . "_error_{$user->ID}";
 		$success_transient_key = Config::PASSWORD_DETECTION_TRANSIENT_PREFIX . "_success_{$user->ID}";
 
-		$error_data   = get_transient( $error_transient_key );
-		$success_data = get_transient( $success_transient_key );
+		$error_data   = $this->extract_and_clear_transient_data( $error_transient_key );
+		$success_data = $this->extract_and_clear_transient_data( $success_transient_key );
 
-		delete_transient( $error_transient_key );
-		delete_transient( $success_transient_key );
+		$interim_login_success = $success_data['code'] === 'auth_code_success' ? 'interim-login-success' : '';
+		$body_classes          = implode( ' ', array_filter( array( 'password-detection-wrapper', $interim_login_success ) ) );
```

Comment:
This could be simplified a bit:

```javascript
$body_class_name = 'password-detection-wrapper';
if ( 'auth_code_success' === $success_data[ 'code' ] ) {
	$body_class_name .= ' interim-login-success';
}
```

```php
<body class="<?php echo $body_class_name; ?>">
```

I don't think the `esc_attr` is necessary in this case as we control the data being inserted.

Reply:
Simplification is nice, unfortunately the linter yells if any output isn't wrapped in an escaping function.